utils/resource/usage.go

```diff
@@ -150,6 +155,22 @@ func (m *manager) TrackProcess(pid int) {
 	m.processesLock.Unlock()
 }
 
+func (m *manager) HasProcess(pid int) bool {
+	m.processesLock.Lock()
+	defer m.processesLock.Unlock()
+	proc, ok := m.processes[pid]
+	if !ok {
+		return false
+	}
+	p, err := os.FindProcess(int(proc.p.Pid))
```

Comment:
How confident are we that this works on all architectures?

Reply:
honestly untested I can see its tested in unix and requires some special handling for windows, unsupported in plan9.

Reply:
I think instead of checking the process I am going to do send a gRPC health RPC and fail fast.